### PR TITLE
Redux compatible

### PR DESCRIPTION
**Pull request by Wroud**
Implement redux subscription tests, make subscribers immutable. #13

### CODE REVIEW

File: packages/core/src/__tests__/redux/store subscribe.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+          {
+            id: 1,
+            text: 'Hello',
+          },
+        ])
+        done()
+      })
+      store.dispatch(action({ id: 1, text: 'Hello' }))
+    })
+    // test('does not leak private listeners array', done => {
```
Comment (belozer) [inline packages/core/src/__tests__/redux/store subscribe.ts]: @artalar maybe need skip tests (`test.skip`)?

File: packages/core/src/__tests__/redux/atom subscribe.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+          {
+            id: 1,
+            text: 'Hello',
+          },
+        ])
+        done()
+      })
+      store.dispatch(action({ id: 1, text: 'Hello' }))
+    })
+    // test('does not leak private listeners array', done => {
```
Comment (belozer) [inline packages/core/src/__tests__/redux/atom subscribe.ts]: @artalar maybe need skip tests (test.skip)?

**Comment by artalar** [comment]
@all-contributors please add @Wroud  for code, ideas, infra, tests, platform, plugin, tool

File: packages/core/src/createStore.ts [hunk 3]
```diff
@@ -51,6 +53,21 @@ export function createStore(
...
+    if (nextActionsListiners === actionsListeners) {
+      nextActionsListiners = actionsListeners.slice()
+    }
+  }
+
+  function ensureCanMutateNextAtomsListeners() {
+    if (nextAtomsListeners === atomsListeners) {
+      nextAtomsListeners = new Map<TreeId, Function[]>()
+      for (const [key, value] of atomsListeners) {
+        nextAtomsListeners.set(key, [...value])
```
Comment (artalar) [inline packages/core/src/createStore.ts]: 1) `if (nextAtomsListeners.get(key) === atomsListeners.get(key))`
2) `[...value]` => `value.slice(0)`

File: packages/core/src/createStore.ts [hunk 4]
```diff
@@ -67,17 +84,25 @@ export function createStore(
   }
 
   // @ts-ignore
-  function subscribe(target, cb) {
-    const isActionSubscription = cb === undefined
-    const listener = safetyFunc(isActionSubscription ? target : cb, 'listener')
+  function subscribe<T>(
+    target: Atom<T> | DispatchFunction,
+    subscriber: (state: T) => any,
```
Comment (artalar) [inline packages/core/src/createStore.ts]: Why not optional?

**Pull request merged**